### PR TITLE
Sharing permission

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Comment by TheOneRing** [comment]
https://jenkins.int.owncloud.com/job/client-trigger-drone/11/

**Comment by TheOneRing** [comment]
Please test https://jenkins.int.owncloud.com/job/client-trigger-drone/14/

File: src/gui/shareusergroupwidget.cpp [hunk 1]
```diff
@@ -288,17 +288,19 @@ void ShareUserGroupWidget::slotCompleterActivated(const QModelIndex &index)
      * https://github.com/owncloud/core/issues/22122#issuecomment-185637344
      * https://github.com/owncloud/client/issues/4996
      */
+    const SharePermissions defaultPermissions = static_cast<SharePermissions>(_account->capabilities().defaultPermissions());
     if (sharee->type() == Sharee::Federated
         && _account->serverVersionInt() < Account::makeServerVersion(9, 1, 0)) {
-        int permissions = SharePermissionRead | SharePermissionUpdate;
+        SharePermissions permissions = SharePermissionRead | SharePermissionUpdate;
```
Comment (dschmidt) [inline src/gui/shareusergroupwidget.cpp:294]: Could you use auto here?

File: src/gui/shareusergroupwidget.cpp [hunk 2]
```diff
@@ -593,35 +595,21 @@ QSharedPointer<Share> ShareUserLine::share() const
...
-    }
+    _permissionUpdate->setChecked(perm & SharePermissionUpdate);
+    _permissionCreate->setChecked(!_isFile && perm & SharePermissionCreate);
+    _permissionDelete->setChecked(!_isFile && perm & SharePermissionDelete);
 
     if (perm & SharePermissionUpdate
-        && (_isFile
-               || (perm & SharePermissionCreate
-                      && perm & SharePermissionDelete))) {
+        && (_isFile || perm & (SharePermissionCreate | SharePermissionDelete))) {
```
Comment (dschmidt) [inline src/gui/shareusergroupwidget.cpp:605]: The logic changed here, please clarify

**Pull request merged**